core/src/test/java/org/apache/calcite/test/JdbcTest.java

```diff
@@ -5898,17 +5965,32 @@ private void checkMapSchemaDirectConnection(String s) throws SQLException {
             + "order by \"empid\" limit 2", null);
     with
         .query("select \"name\" from \"adhoc\".V order by \"name\"")
+        .withHook(Hook.SQL2REL_CONVERTER_CONFIG_BUILDER,
+            (Consumer<Holder<Config>>) configHolder ->
+                configHolder.set(configHolder.get().withRemoveSortInSubQuery(false)))
```

Comment:
I don't understand why this hook is still needed. The view is not top-level, so its results should not be sorted, but the LIMIT on the sorted data should still apply. Does the test succeed if you remove the hook?

Reply:
The new code doesn't need a hook.